compiler/func.go

```diff
@@ -13,14 +10,85 @@ import (
 	"tinygo.org/x/go-llvm"
 )
 
+type funcValueImplementation int
+
+const (
+	funcValueNone funcValueImplementation = iota
+
+	// A func value is implemented as a pair of pointers:
+	//     {context, function pointer}
+	// where the context may be a pointer to a heap-allocated struct containing
+	// the free variables, or it may be undef if the function being pointed to
+	// doesn't need a context. The function pointer is a regular function
+	// pointer.
+	funcValueDoubleword
+
+	// As funcValueDoubleword, but with the function pointer replaced by a
+	// unique ID per function signature. Function values are called by using a
+	// switch statement and choosing which function to call.
+	funcValueSwitch
+)
+
+func (c *Compiler) funcImplementation() funcValueImplementation {
+	return funcValueSwitch
```

Comment:
Oops, I left something here for testing. Will push a fix for this.

Reply:
Don't forget about this. :)

Reply:
Done.